### ISSUE TITLE
XFA - missing content on Ontario Lease

### ISSUE THREAD

**Issue by jrmuizel** [mozilla/pdf.js#13611]
Attach (recommended) or Link to PDF file here:
http://www.forms.ssb.gov.on.ca/mbs/ssb/forms/ssbforms.nsf/GetFileAttach/047-2229E~1/$File/2229E.pdf

Configuration:
- Web browser and its version: FF 91 (2021-06-21) with XFA enabled
- Operating system and its version: macOS
- PDF.js version: 2.10.146
- Is a browser extension: Yes

Steps to reproduce the problem:
1. Load the PDF and compare to Adobe Reader or Chrome Canary with XFA turned on. (Chrome has lots of blank pages but seems to draw more stuff)

**Comment by marco-c** [comment]
[2229E.pdf](https://github.com/mozilla/pdf.js/files/6695365/2229E.pdf)

**Comment by marco-c** [comment]
This looks like a dynamic form.

**Comment by calixteman** [comment]
Yes it's a dynamic form but likely something is wrong in the layout: I'll have a look on it.

**Comment by jrmuizel** [comment]
This PDF seems to draw correctly now. Most of the form controls are unusable though.

**Comment by jrmuizel** [comment]
This seems to work well enough now. Certainly much better than Chrome Canary

**Comment by marco-c** [comment]
Let's keep this open for the JS parts.

**Comment by jrmuizel** [comment]
What's the status of the JS stuff?

**Comment by calixteman** [comment]
For now: there are no plan to implement JS stuff but it's something we likely consider in the future.
From my pov, it will be a bit challenging because thanks JS, layout will become dynamic (it's possible to add pages, remove pages, add some text in a place which will move a lot of elements on the next page....).
So compared to what we did for classic Acroforms, my guess is that it'll be "slightly" harder.
